### ISSUE TITLE
Support Extension on Firefox

### ISSUE THREAD

**Issue by u254254** [csfloat/extension#243]
Popup "Setup Trade Verification" says this:
> At the moment, you need to use a Chromium browser (ie. Chrome, Opera GX) in order to use the CSFloat extension. In the future, Firefox will support it -- sorry!

Version 4.3.1 for Firefox is already on GitHub Releases, meaning it *technically* supports Firefox. Why not just submit new version to Mozilla and make it available in Firefox?

**Comment by Fesiug** [comment]
Agreed, having to install and use a Chromium browser side-by-side with my normal one to verify trades is annoying. What's keeping the newest version extension from being updated and pushed to Firefox Addons?

**Comment by Step7750** [comment]
Current blocker for the Firefox extension is mainly related to permission handling. In FF land, permissions specified by the manifest are still treated somewhat "optional" by the browser -- which completely breaks functionality (ie. annoying support tickets on our end when users fiddle around).

The current dialog that grants permissions requires a trusted event to propagate to the service worker in order to request permissions. In Chrome, it properly handles scoping the closure with "trust", but Firefox has a long-standing issue where this is not the case. It'd require reworking where/how to grant permissions since requesting the user to manually grant it by going to the extension's settings is very error prone.

Given these issues, we've focused on other platforms such as mobile for the time being.

**Comment by Step7750** [comment]
> Version 4.3.1 for Firefox is already on GitHub Releases, meaning it technically supports Firefox. Why not just submit new version to Mozilla and make it available in Firefox?

These versions are automatically built by a bot, Firefox doesn't support trade offer tracking due to the reasons mentioned above.
